src/System.Reflection/tests/MethodInfo/MethodInfo_PropertyTests.cs

```diff
@@ -13,191 +13,101 @@ namespace System.Reflection.Tests
         {
-            MethodInfo mi = GetMethod("DummyMethod5");
-
-            Assert.True(mi.IsVirtual);
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod5").IsVirtual, true};
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsVirtual, false};
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsPublic, true };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsPrivate, false };
+            yield return new object[] { GetMethod(typeof(classA), "method1").IsStatic, true };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsStatic, false };
+            yield return new object[] { GetMethod(typeof(classA), "GenericMethod").IsGenericMethod, true};
+            yield return new object[] { GetMethod(typeof(classA), "method1").IsGenericMethod, false };
+            yield return new object[] { GetMethod(typeof(classA), "method1").IsGenericMethodDefinition, false };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsFinal, false };
+            yield return new object[] { GetMethod(typeof(classC), "virtualMethod").IsFinal, true };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsConstructor, false};
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsAbstract, false};
+            yield return new object[] { GetMethod(typeof(classB), "abstractMethod").IsAbstract, true};
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsAssembly, false};
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsFamily, false};
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsFamilyAndAssembly, false };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsFamilyOrAssembly, false };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").ContainsGenericParameters, false};
+            yield return new object[] { GetMethod(typeof(classA), "GenericMethod").ContainsGenericParameters, true };
+            yield return new object[] { GetMethod(typeof(MethodInfoPropertyTests), "DummyMethod1").IsSpecialName, false};
+            yield return new object[] { GetMethod(typeof(classC), "abstractMethod").IsHideBySig, true};
         }
 
-
-        //Verify IsVirtual Property
-        [Fact]
-        public static void TestIsVirtual2()
+        //Verify Properties
+        [Theory]
+        [MemberData(nameof(TestPropertyData))]
+        public static void TestProperty(bool actual, bool expected)
```

Comment:
FWIW, I find the organization of this test a bit confusing.  I'm having trouble understanding what's being tested and compared to what.  For this one, it might actually be better to not use a Theory at all.  You could just use a Fact that has a bunch of asserts in it, and potentially split it up into several tests.